allennlp/nn/beam_search.py

```diff
@@ -408,6 +443,10 @@ def _search(
         # shape: (batch_size, beam_size, max_steps)
         all_predictions = torch.cat(list(reversed(reconstructed_predictions)), 2)
 
+        # If stochastic beam search, return the true probabilities instead of perturbed
+        if isinstance(self.sampler, GumbelMaxSampler):
+            return all_predictions, last_true_probabilities
+
         return all_predictions, last_log_probabilities
```

Comment:
With the above changes, we'd just always return `last_true_log_probabilities`.

```suggestion
        return all_predictions, last_true_log_probabilities
```